lib/lists.nix

```diff
@@ -35,13 +40,25 @@ rec {
         else op (elemAt list n) (fold' (n + 1));
     in fold' 0;
 
-  /* Left fold: `fold op nul [x_1 x_2 ... x_n] == op (... (op (op nul
-     x_1) x_2) ... x_n)'.
+  /* fold is an alias of foldr for historic reasons */
+  # FIXME(Profpatsch): deprecate?
+  fold = foldr;
```

Comment:
Nitpick: I expect you meant to write commutative not associative?

I think the current docs explain which to prefer quite well: use (strict) foldl if you intend to consume the entire result right away, esp. when reducing to atomic values, otherwise you probably want to use foldr.